pkg/didcomm/common/service/did_comm_msg.go

```diff
@@ -134,3 +140,78 @@ func (m DIDCommMsgMap) Decode(v interface{}) error {
 
 	return decoder.Decode(m)
 }
+
+func toMap(v interface{}) map[string]interface{} {
+	res := make(map[string]interface{})
+
+	rv := reflect.Indirect(reflect.ValueOf(v))
```

Comment:
some comments within these functions could be helpful, particularly for those who do not use reflect much.

Reply:
added more comments